samples/web/content/apprtc/apprtc.py

```diff
@@ -495,12 +526,115 @@ def write_response(self, result, params, messages):
       'result': result,
       'params': params
     }))
+    
+  def report_error(self, error):
+    self.write_response(error, {}, [])
 
   def write_room_parameters(self, room_id, client_id, messages, is_initiator):
     params = get_room_parameters(self.request, room_id, client_id, is_initiator)
     self.write_response('SUCCESS', params, messages)
+    
+  def handle_call(self, msg, room_id):
+    """Handles a new call request."""
+    if not util.has_msg_fields(msg, ((PARAM_CALLER_GCM_ID, basestring),
+        (PARAM_CALLEE_ID, basestring))):
+      self.report_error(constants.RESPONSE_INVALID_ARGUMENT)
+      return
+      
+    caller_gcm_id = msg[PARAM_CALLER_GCM_ID]
+    callee_id = msg[PARAM_CALLEE_ID]
+    # Look up and validate caller by gcm id.
+    # TODO (chuckhays): Once registration is enabled, turn on restriction
+    # to only return verified records.
+    caller_records = gcmrecord.GCMRecord.get_by_gcm_id(caller_gcm_id, False)
+    if len(caller_records) < 1:
+      self.report_error(constants.RESPONSE_INVALID_CALLER)
+      return
+    caller_id = caller_records[0].user_id
```

Comment:
Seems reasonable, since it now supports a list I can just add all of the callee's gcm ids (and later only verified ones)